### PR TITLE
Configurable commitWithin behaviour

### PR DESCRIPTION
**Pull request by serggl**
This enables Solr's [commitWithin](https://wiki.apache.org/solr/CommitWithin) feature and allows to configure commit interval via `sunspot.yml`

Same behavior could be achieved by configuring `autoSoftCommit` property of `solrconfig.xml`, but this requires server restart which is sometimes extremely expensive (like I'm my case). 
So  having an option to configure this feature via Rails app could be useful.

P.S. don't see a good place to describe this in README.md, so Im open to this if someone can point me to a proper section

### CODE REVIEW

**Comment by serggl** [comment]
@bragboy can you please review this PR? thanks!

File: sunspot/lib/sunspot/indexer.rb [hunk 1]
```diff
@@ -9,8 +9,9 @@ module Sunspot
   #
   class Indexer #:nodoc:
 
-    def initialize(connection)
+    def initialize(connection, options = {})
       @connection = connection
+      @update_options = {params: options.delete_if { |key, value| value.nil? }}
```
Comment (JoniAmir) [inline sunspot/lib/sunspot/indexer.rb]: Codacy Issue found: [Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.](https://www.codacy.com/app/joni-amir/sunspot/file/2482066248/issues/source?bid=3265104&fileBranchId=3265120#l14)

File: sunspot/spec/mocks/connection.rb [hunk 2]
```diff
@@ -124,5 +128,9 @@ def with?(request, params)
         request.has_key?(params)
       end
     end
+
+    def commitWithinValue(opts = {})
```
Comment (JoniAmir) [inline sunspot/spec/mocks/connection.rb]: Codacy Issue found: [Use snake_case for method names.](https://www.codacy.com/app/joni-amir/sunspot/file/2482066397/issues/source?bid=3265104&fileBranchId=3265120#l132)

**Comment by serggl** [comment]
@JoniAmir code-style issues were fixed

**Comment by JoniAmir** [comment]
👍

File: sunspot/lib/sunspot/indexer.rb [hunk 3]
```diff
@@ -11,7 +11,7 @@ class Indexer #:nodoc:
 
     def initialize(connection, options = {})
       @connection = connection
-      @update_options = {params: options.delete_if { |key, value| value.nil? }}
+      @update_options = {params: options.delete_if { |_key, value| value.nil? }}
```
Comment (YoniAmir) [inline sunspot/lib/sunspot/indexer.rb]: Not `_key` but rather `_`

**Comment by serggl** [comment]
@bragboy  any updates on this? is it going to be accepted or rejected?

**Pull request closed without merging**